x-pack/test/apm_api_integration/tests/mobile/mobile_main_statistics_by_field.spec.ts

```diff
@@ -0,0 +1,143 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { ENVIRONMENT_ALL } from '@kbn/apm-plugin/common/environment_filter_values';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { generateMobileData } from './generate_mobile_data';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const apmApiClient = getService('apmApiClient');
+  const registry = getService('registry');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  const start = new Date('2023-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2023-01-01T00:15:00.000Z').getTime() - 1;
+
+  async function getMobileMainStatisticsByField({
+    environment = ENVIRONMENT_ALL.value,
+    kuery = '',
+    serviceName,
+    field,
+  }: {
+    environment?: string;
+    kuery?: string;
+    serviceName: string;
+    field: string;
+  }) {
+    return await apmApiClient
+      .readUser({
+        endpoint: 'GET /internal/apm/mobile-services/{serviceName}/main_statistics',
+        params: {
+          path: { serviceName },
+          query: {
+            environment,
+            start: new Date(start).toISOString(),
+            end: new Date(end).toISOString(),
+            kuery,
+            field,
+          },
+        },
+      })
+      .then(({ body }) => body);
+  }
+
+  registry.when(
+    'Mobile main statistics when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      describe('when no data', () => {
+        it('handles empty state', async () => {
+          const response = await getMobileMainStatisticsByField({
+            serviceName: 'foo',
+            field: 'service.version',
+          });
+          expect(response.mainStatistics.length).to.be(0);
+        });
+      });
+    }
+  );
+
+  registry.when('Mobile main statistics', { config: 'basic', archives: [] }, () => {
+    before(async () => {
+      await generateMobileData({
+        synthtraceEsClient,
+        start,
+        end,
+      });
+    });
+
+    after(() => synthtraceEsClient.clean());
+
+    describe('when data is loaded', () => {
+      it('returns the correct data for App version', async () => {
+        const response = await getMobileMainStatisticsByField({
+          serviceName: 'synth-android',
+          environment: 'production',
+          field: 'service.version',
+        });
+        const fieldValues = response.mainStatistics.map((item) => item.name);
+
+        expect(fieldValues).to.be.eql(['2.3', '1.1', '1.2']);
+
+        const latencyValues = response.mainStatistics.map((item) => item.latency);
+
+        expect(latencyValues).to.be.eql([172000, 20000, 20000]);
+
+        const throughputValues = response.mainStatistics.map((item) => item.throughput);
+        expect(throughputValues).to.be.eql([
+          1.2000013333348147, 0.40000044444493826, 0.40000044444493826,
+        ]);
```

Comment:
This is more like snapshot testing. How do you calculate these values based on the inputs for generating data? I'd like to see that calculation in this file instead of writing out numbers that could as well be random to the reader.